src/System.Net.Security/src/System/Net/NegotiateStreamPal.Windows.cs

```diff
@@ -0,0 +1,458 @@
+                    }
+
+                    Debug.Fail("NTAuthentication#" + LoggingHash.HashString(securityContext) + "::Encrypt", "Arguments out of range.");
+                }
+
+                throw;
+            }
+
+            int resultSize = count + sizes.SecurityTrailer + sizes.BlockSize;
+            if (output == null || output.Length < resultSize + 4)
+            {
+                output = new byte[resultSize + 4];
+            }
+
+            // Make a copy of user data for in-place encryption.
+            Buffer.BlockCopy(buffer, offset, output, 4 + sizes.SecurityTrailer, count);
+
+            // Prepare buffers TOKEN(signature), DATA and Padding.
+            var securityBuffer = new SecurityBuffer[3];
+            securityBuffer[0] = new SecurityBuffer(output, 4, sizes.SecurityTrailer, SecurityBufferType.Token);
+            securityBuffer[1] = new SecurityBuffer(output, 4 + sizes.SecurityTrailer, count, SecurityBufferType.Data);
+            securityBuffer[2] = new SecurityBuffer(output, 4 + sizes.SecurityTrailer + count, sizes.BlockSize, SecurityBufferType.Padding);
+
+            int errorCode;
+            if (isConfidential)
+            {
+                errorCode = SSPIWrapper.EncryptMessage(GlobalSSPI.SSPIAuth, securityContext, securityBuffer, sequenceNumber);
+            }
+            else
+            {
+                if (isNtlm)
+                {
+                    securityBuffer[1].type |= SecurityBufferType.ReadOnlyFlag;
+                }
+
+                errorCode = SSPIWrapper.MakeSignature(GlobalSSPI.SSPIAuth, securityContext, securityBuffer, 0);
+            }
+
+            if (errorCode != 0)
+            {
+                if (GlobalLog.IsEnabled)
+                {
+                    GlobalLog.Print("NTAuthentication#" + LoggingHash.HashString(securityContext) + "::Encrypt() throw Error = " + errorCode.ToString("x", NumberFormatInfo.InvariantInfo));
+                }
+                throw new Win32Exception(errorCode);
+            }
+
+            // Compacting the result.
+            resultSize = securityBuffer[0].size;
+            bool forceCopy = false;
+            if (resultSize != sizes.SecurityTrailer)
+            {
+                forceCopy = true;
+                Buffer.BlockCopy(output, securityBuffer[1].offset, output, 4 + resultSize, securityBuffer[1].size);
+            }
+
+            resultSize += securityBuffer[1].size;
+            if (securityBuffer[2].size != 0 && (forceCopy || resultSize != (count + sizes.SecurityTrailer)))
+            {
+                Buffer.BlockCopy(output, securityBuffer[2].offset, output, 4 + resultSize, securityBuffer[2].size);
+            }
+
+            resultSize += securityBuffer[2].size;
+            unchecked
+            {
+                // TODO (Issue #6063): Should be offset by offset
```

Comment:
@shrutigarg Without a lot of context, I don't understand what this means. 

Please add more information in the comment. I've reactivated #6063: this comment still exists today but the issue was closed.